src/main/Backend.java

```diff
@@ -7,141 +7,80 @@
 import com.ericsson.otp.erlang.OtpErlangBinary;
 import com.ericsson.otp.erlang.OtpErlangAtom;
 import com.ericsson.otp.erlang.OtpErlangPid;
-import com.ericsson.otp.erlang.OtpErlangString;
 import com.ericsson.otp.erlang.OtpErlangTuple;
 import com.ericsson.otp.erlang.OtpErlangObject;
-import com.ericsson.otp.erlang.OtpMbox;
-import com.ericsson.otp.erlang.OtpNode;
 
-public class Backend {
-    OtpMbox myOtpMbox = null;
-    OtpNode myOtpNode = null;
-    OtpErlangPid last_pid;
+public class Backend extends AntidoteBackend {
 
     Map<OtpErlangBinary, CRDT> ObjectTable = new Hashtable<>();
 
-    public Backend(String NodeName, String MailBox) {
-        try {
-            myOtpNode = new OtpNode(NodeName);
-            System.out.println(myOtpNode);
-            myOtpMbox = myOtpNode.createMbox();
-            myOtpMbox.registerName(MailBox);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+    public Backend(String NodeName, String MailBox, String cookie) {
+        super(NodeName, MailBox, cookie);
     }
 
-    public Backend(String NodeName, String MailBox, String cookie) {
-        try {
-            // KEEP THIS
-            // If the next line raises an exception, you need to get empd running
-            // Do something like `erl -sname whatever` to check and get it running if not
-            // KEEP THIS
-            myOtpNode = new OtpNode(NodeName);
-            System.out.println(myOtpNode);
-            myOtpNode.setCookie(cookie);
-            myOtpMbox = myOtpNode.createMbox();
-            myOtpMbox.registerName(MailBox);
-        } catch (Exception e) {
-            e.printStackTrace();
+    public OtpErlangBinary value(OtpErlangBinary JavaObjectId) throws NoSuchObjectException {
+        // assert that object is in table else request it
+        CRDT crdt_object = ObjectTable.get(JavaObjectId);
+        if (crdt_object == null) {
+            throw new NoSuchObjectException();
         }
+        return new OtpErlangBinary(crdt_object.read());
     }
 
-    public void run(String target) {
-        if (myOtpNode.ping(target, 2000)) {
-            System.out.println("remote is up");
+    // Should this be split up into two, one for a crdt binary and one for a
+    // GenericFunction binary?
+    public OtpErlangBinary update(OtpErlangBinary JavaObjectId, OtpErlangBinary binary) throws NoSuchObjectException {
+        if (!ObjectTable.containsKey(JavaObjectId)) {
+            CRDT thing;
+            try {
+                thing = (CRDT) binary.getObject();
+            } catch (ClassCastException e) {
+                throw new NoSuchObjectException();
```

Comment:
will this always be resolved by the "just call get" logic, or are there other reasons to get a ClassCastException?

Reply:
Put a comment in that the other case is this is a generic function -- and probably add an assert to verify this fact.

Reply:
assert(hasBeenResolved());